vm/proxyapp: `TestCtor_TCP_Reconnect_On_LostConnection` timed out

```
panic: test timed out after 10m0s
	running tests:
		TestCtor_TCP_Reconnect_On_LostConnection (10m0s)

goroutine 13640658 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:2484 +0x394
created by time.goFunc
	/usr/local/go/src/time/sleep.go:215 +0x2d

goroutine 1 [chan receive, 9 minutes]:
testing.(*T).Run(0xc000b03500, {0x161155d?, 0xc0007adb30?}, 0x1f544a8)
	/usr/local/go/src/testing/testing.go:1859 +0x431
testing.runTests.func1(0xc000b03500)
	/usr/local/go/src/testing/testing.go:2279 +0x37
testing.tRunner(0xc000b03500, 0xc0007adc70)
	/usr/local/go/src/testing/testing.go:1792 +0xf4
testing.runTests(0xc000b22330, {0x2e09b40, 0x1f, 0x1f}, {0x2e2aae0?, 0x7?, 0x2e28b60?})
	/usr/local/go/src/testing/testing.go:2277 +0x4b4
testing.(*M).Run(0xc000bace60)
	/usr/local/go/src/testing/testing.go:2142 +0x64a
main.main()
	_testmain.go:105 +0x9b

goroutine 12 [chan receive]:
github.com/google/syzkaller/pkg/stat.newSet.func1()
	/syzkaller/gopath/src/github.com/google/syzkaller/pkg/stat/set.go:95 +0x66
created by github.com/google/syzkaller/pkg/stat.newSet in goroutine 1
	/syzkaller/gopath/src/github.com/google/syzkaller/pkg/stat/set.go:94 +0xd6

goroutine 203 [IO wait, 10 minutes]:
internal/poll.runtime_pollWait(0x7f599aed65f0, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc000be6180?, 0x900000036?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Accept(0xc000be6180)
	/usr/local/go/src/internal/poll/fd_unix.go:620 +0x295
net.(*netFD).accept(0xc000be6180)
	/usr/local/go/src/net/fd_unix.go:172 +0x29
net.(*TCPListener).accept(0xc000b27680)
	/usr/local/go/src/net/tcpsock_posix.go:159 +0x1b
net.(*TCPListener).Accept(0xc000b27680)
	/usr/local/go/src/net/tcpsock.go:380 +0x30
github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener.func1()
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:210 +0x5f
created by github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener in goroutine 202
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:208 +0x291

goroutine 149 [select]:
github.com/google/syzkaller/vm/proxyapp.(*ProxyApp).doLogPooling.func1()
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:324 +0x11e
created by github.com/google/syzkaller/vm/proxyapp.(*ProxyApp).doLogPooling in goroutine 202
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:314 +0xec

goroutine 148 [IO wait]:
internal/poll.runtime_pollWait(0x7f599aed6a50, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc000be6880?, 0xc000a92601?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc000be6880, {0xc000a92601, 0x5ff, 0x5ff})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x27a
net.(*netFD).Read(0xc000be6880, {0xc000a92601?, 0x412d05?, 0xc297c27d48?})
	/usr/local/go/src/net/fd_posix.go:55 +0x25
net.(*conn).Read(0xc000a32030, {0xc000a92601?, 0x412ad4?, 0xc19daece38?})
	/usr/local/go/src/net/net.go:194 +0x45
encoding/json.(*Decoder).refill(0xc000a3e000)
	/usr/local/go/src/encoding/json/stream.go:165 +0x188
encoding/json.(*Decoder).readValue(0xc000a3e000)
	/usr/local/go/src/encoding/json/stream.go:140 +0x85
encoding/json.(*Decoder).Decode(0xc000a3e000, {0x12f2a40, 0xc000a120c8})
	/usr/local/go/src/encoding/json/stream.go:63 +0x75
net/rpc/jsonrpc.(*clientCodec).ReadResponseHeader(0xc000a12080, 0xc0005f7710)
	/usr/local/go/src/net/rpc/jsonrpc/client.go:76 +0x65
net/rpc.(*Client).input(0xc000a162a0)
	/usr/local/go/src/net/rpc/client.go:109 +0x9f
created by net/rpc.NewClientWithCodec in goroutine 202
	/usr/local/go/src/net/rpc/client.go:206 +0xb6

goroutine 1606 [chan receive, 9 minutes]:
github.com/google/syzkaller/vm/proxyapp.TestCtor_TCP_Reconnect_On_LostConnection(0xc000be4000?)
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:158 +0x2fb
testing.tRunner(0xc000be4000, 0x1f544a8)
	/usr/local/go/src/testing/testing.go:1792 +0xf4
created by testing.(*T).Run in goroutine 1
	/usr/local/go/src/testing/testing.go:1851 +0x413

goroutine 108 [IO wait]:
internal/poll.runtime_pollWait(0x7f59996125f8, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc000650280?, 0xc000a92001?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc000650280, {0xc000a92001, 0x5ff, 0x5ff})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x27a
net.(*netFD).Read(0xc000650280, {0xc000a92001?, 0xc19dc2f150?, 0xc19dc2f150?})
	/usr/local/go/src/net/fd_posix.go:55 +0x25
net.(*conn).Read(0xc00011e960, {0xc000a92001?, 0x1?, 0x11?})
	/usr/local/go/src/net/net.go:194 +0x45
encoding/json.(*Decoder).refill(0xc00065bb80)
	/usr/local/go/src/encoding/json/stream.go:165 +0x188
encoding/json.(*Decoder).readValue(0xc00065bb80)
	/usr/local/go/src/encoding/json/stream.go:140 +0x85
encoding/json.(*Decoder).Decode(0xc00065bb80, {0x12f29e0, 0xc0001493a0})
	/usr/local/go/src/encoding/json/stream.go:63 +0x75
net/rpc/jsonrpc.(*serverCodec).ReadRequestHeader(0xc000149380, 0xc000538680)
	/usr/local/go/src/net/rpc/jsonrpc/server.go:66 +0x6c
net/rpc.(*Server).readRequestHeader(0xc000be0550, {0x20c43c8, 0xc000149380})
	/usr/local/go/src/net/rpc/server.go:587 +0x58
net/rpc.(*Server).readRequest(0xc000be0550?, {0x20c43c8, 0xc000149380})
	/usr/local/go/src/net/rpc/server.go:547 +0x2f
net/rpc.(*Server).ServeCodec(0xc000be0550, {0x20c43c8, 0xc000149380})
	/usr/local/go/src/net/rpc/server.go:462 +0x71
created by github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener.func1 in goroutine 203
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:214 +0x2b2

goroutine 111 [select, 10 minutes]:
github.com/google/syzkaller/vm/proxyapp.ctor.func1()
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:59 +0x13c
created by github.com/google/syzkaller/vm/proxyapp.ctor in goroutine 202
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:47 +0x1c5

goroutine 1512 [IO wait]:
internal/poll.runtime_pollWait(0x7f599aed6b68, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc000c58b00?, 0xc0009eaa80?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc000c58b00, {0xc0009eaa80, 0xa80, 0xa80})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x27a
net.(*netFD).Read(0xc000c58b00, {0xc0009eaa80?, 0xc19db6f680?, 0xc19dd0226e?})
	/usr/local/go/src/net/fd_posix.go:55 +0x25
net.(*conn).Read(0xc0008bc0b8, {0xc0009eaa80?, 0x7f5838cec4e8?, 0x7f59e22c15c0?})
	/usr/local/go/src/net/net.go:194 +0x45
crypto/tls.(*atLeastReader).Read(0xc24c52fc80, {0xc0009eaa80?, 0x1275040?, 0x1?})
	/usr/local/go/src/crypto/tls/conn.go:809 +0x3b
bytes.(*Buffer).ReadFrom(0xc0005157b8, {0x20a4ce0, 0xc24c52fc80})
	/usr/local/go/src/bytes/buffer.go:211 +0x98
crypto/tls.(*Conn).readFromUntil(0xc000515508, {0x20a4f00, 0xc0008bc0b8}, 0x4424d4?)
	/usr/local/go/src/crypto/tls/conn.go:831 +0xde
crypto/tls.(*Conn).readRecordOrCCS(0xc000515508, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:629 +0x3cf
crypto/tls.(*Conn).readRecord(...)
	/usr/local/go/src/crypto/tls/conn.go:591
crypto/tls.(*Conn).Read(0xc000515508, {0xc000b3a001, 0x5ff, 0xc19db6f6a8?})
	/usr/local/go/src/crypto/tls/conn.go:1385 +0x145
encoding/json.(*Decoder).refill(0xc000aa4780)
	/usr/local/go/src/encoding/json/stream.go:165 +0x188
encoding/json.(*Decoder).readValue(0xc000aa4780)
	/usr/local/go/src/encoding/json/stream.go:140 +0x85
encoding/json.(*Decoder).Decode(0xc000aa4780, {0x12f2a40, 0xc000c58e48})
	/usr/local/go/src/encoding/json/stream.go:63 +0x75
net/rpc/jsonrpc.(*clientCodec).ReadResponseHeader(0xc000c58e00, 0xc000b25860)
	/usr/local/go/src/net/rpc/jsonrpc/client.go:76 +0x65
net/rpc.(*Client).input(0xc000a17800)
	/usr/local/go/src/net/rpc/client.go:109 +0x9f
created by net/rpc.NewClientWithCodec in goroutine 112
	/usr/local/go/src/net/rpc/client.go:206 +0xb6

goroutine 1511 [IO wait]:
internal/poll.runtime_pollWait(0x7f599aed6eb0, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc000c58c00?, 0xc000996800?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc000c58c00, {0xc000996800, 0x800, 0x800})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x27a
net.(*netFD).Read(0xc000c58c00, {0xc000996800?, 0x9?, 0x5cf?})
	/usr/local/go/src/net/fd_posix.go:55 +0x25
net.(*conn).Read(0xc0008bc0b0, {0xc000996800?, 0x7f5838ce9248?, 0x7f59e22c3210?})
	/usr/local/go/src/net/net.go:194 +0x45
crypto/tls.(*atLeastReader).Read(0xc24c373260, {0xc000996800?, 0x7fb?, 0xc298101908?})
	/usr/local/go/src/crypto/tls/conn.go:809 +0x3b
bytes.(*Buffer).ReadFrom(0xc000515438, {0x20a4ce0, 0xc24c373260})
	/usr/local/go/src/bytes/buffer.go:211 +0x98
crypto/tls.(*Conn).readFromUntil(0xc000515188, {0x20a4f00, 0xc0008bc0b0}, 0x4424d4?)
	/usr/local/go/src/crypto/tls/conn.go:831 +0xde
crypto/tls.(*Conn).readRecordOrCCS(0xc000515188, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:629 +0x3cf
crypto/tls.(*Conn).readRecord(...)
	/usr/local/go/src/crypto/tls/conn.go:591
crypto/tls.(*Conn).Read(0xc000515188, {0xc0008fe001, 0x5ff, 0x11?})
	/usr/local/go/src/crypto/tls/conn.go:1385 +0x145
encoding/json.(*Decoder).refill(0xc000aa43c0)
	/usr/local/go/src/encoding/json/stream.go:165 +0x188
encoding/json.(*Decoder).readValue(0xc000aa43c0)
	/usr/local/go/src/encoding/json/stream.go:140 +0x85
encoding/json.(*Decoder).Decode(0xc000aa43c0, {0x12f29e0, 0xc0006cc200})
	/usr/local/go/src/encoding/json/stream.go:63 +0x75
net/rpc/jsonrpc.(*serverCodec).ReadRequestHeader(0xc0006cc1e0, 0xc0008b3d40)
	/usr/local/go/src/net/rpc/jsonrpc/server.go:66 +0x6c
net/rpc.(*Server).readRequestHeader(0xc0009980f0, {0x20c43c8, 0xc0006cc1e0})
	/usr/local/go/src/net/rpc/server.go:587 +0x58
net/rpc.(*Server).readRequest(0xc0009980f0?, {0x20c43c8, 0xc0006cc1e0})
	/usr/local/go/src/net/rpc/server.go:547 +0x2f
net/rpc.(*Server).ServeCodec(0xc0009980f0, {0x20c43c8, 0xc0006cc1e0})
	/usr/local/go/src/net/rpc/server.go:462 +0x71
created by github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener.func1 in goroutine 1507
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:214 +0x2b2

goroutine 1513 [select]:
github.com/google/syzkaller/vm/proxyapp.(*ProxyApp).doLogPooling.func1()
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:324 +0x11e
created by github.com/google/syzkaller/vm/proxyapp.(*ProxyApp).doLogPooling in goroutine 112
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:314 +0xec

goroutine 2213 [IO wait]:
internal/poll.runtime_pollWait(0x7f599aed63c0, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc000d86a80?, 0xc000a68601?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc000d86a80, {0xc000a68601, 0x5ff, 0x5ff})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x27a
net.(*netFD).Read(0xc000d86a80, {0xc000a68601?, 0x412d05?, 0xc296d07d48?})
	/usr/local/go/src/net/fd_posix.go:55 +0x25
net.(*conn).Read(0xc000a32000, {0xc000a68601?, 0x412ad4?, 0xc19da7c4a8?})
	/usr/local/go/src/net/net.go:194 +0x45
encoding/json.(*Decoder).refill(0xc00070c000)
	/usr/local/go/src/encoding/json/stream.go:165 +0x188
encoding/json.(*Decoder).readValue(0xc00070c000)
	/usr/local/go/src/encoding/json/stream.go:140 +0x85
encoding/json.(*Decoder).Decode(0xc00070c000, {0x12f2a40, 0xc00076b7c8})
	/usr/local/go/src/encoding/json/stream.go:63 +0x75
net/rpc/jsonrpc.(*clientCodec).ReadResponseHeader(0xc00076b780, 0xc00071e780)
	/usr/local/go/src/net/rpc/jsonrpc/client.go:76 +0x65
net/rpc.(*Client).input(0xc000a5e060)
	/usr/local/go/src/net/rpc/client.go:109 +0x9f
created by net/rpc.NewClientWithCodec in goroutine 1606
	/usr/local/go/src/net/rpc/client.go:206 +0xb6

goroutine 1607 [IO wait, 9 minutes]:
internal/poll.runtime_pollWait(0x7f5999612710, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc000d86000?, 0x0?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Accept(0xc000d86000)
	/usr/local/go/src/internal/poll/fd_unix.go:620 +0x295
net.(*netFD).accept(0xc000d86000)
	/usr/local/go/src/net/fd_unix.go:172 +0x29
net.(*TCPListener).accept(0xc000c9a000)
	/usr/local/go/src/net/tcpsock_posix.go:159 +0x1b
net.(*TCPListener).Accept(0xc000c9a000)
	/usr/local/go/src/net/tcpsock.go:380 +0x30
github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener.func1()
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:210 +0x5f
created by github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener in goroutine 1606
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:208 +0x291

goroutine 2214 [select]:
github.com/google/syzkaller/vm/proxyapp.(*ProxyApp).doLogPooling.func1()
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:324 +0x11e
created by github.com/google/syzkaller/vm/proxyapp.(*ProxyApp).doLogPooling in goroutine 1606
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:314 +0xec

goroutine 1507 [IO wait, 9 minutes]:
internal/poll.runtime_pollWait(0x7f599aed62a8, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc000c58080?, 0x900000036?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Accept(0xc000c58080)
	/usr/local/go/src/internal/poll/fd_unix.go:620 +0x295
net.(*netFD).accept(0xc000c58080)
	/usr/local/go/src/net/fd_unix.go:172 +0x29
net.(*TCPListener).accept(0xc0009b8040)
	/usr/local/go/src/net/tcpsock_posix.go:159 +0x1b
net.(*TCPListener).Accept(0xc0009b8040)
	/usr/local/go/src/net/tcpsock.go:380 +0x30
crypto/tls.(*listener).Accept(0xc000ca5020)
	/usr/local/go/src/crypto/tls/tls.go:67 +0x24
github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener.func1()
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:210 +0x5f
created by github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener in goroutine 112
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:208 +0x291

goroutine 1597 [select, 9 minutes]:
github.com/google/syzkaller/vm/proxyapp.ctor.func1()
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:59 +0x13c
created by github.com/google/syzkaller/vm/proxyapp.ctor in goroutine 112
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:47 +0x1c5

goroutine 455 [IO wait]:
internal/poll.runtime_pollWait(0x7f599aed6c80, 0x72)
	/usr/local/go/src/runtime/netpoll.go:351 +0x85
internal/poll.(*pollDesc).wait(0xc000d00480?, 0xc000a68001?, 0x0)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
internal/poll.(*pollDesc).waitRead(...)
	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
internal/poll.(*FD).Read(0xc000d00480, {0xc000a68001, 0x5ff, 0x5ff})
	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x27a
net.(*netFD).Read(0xc000d00480, {0xc000a68001?, 0xc19db76fe0?, 0xc19db76fe0?})
	/usr/local/go/src/net/fd_posix.go:55 +0x25
net.(*conn).Read(0xc000b32000, {0xc000a68001?, 0x1?, 0x11?})
	/usr/local/go/src/net/net.go:194 +0x45
encoding/json.(*Decoder).refill(0xc000578b40)
	/usr/local/go/src/encoding/json/stream.go:165 +0x188
encoding/json.(*Decoder).readValue(0xc000578b40)
	/usr/local/go/src/encoding/json/stream.go:140 +0x85
encoding/json.(*Decoder).Decode(0xc000578b40, {0x12f29e0, 0xc000d06080})
	/usr/local/go/src/encoding/json/stream.go:63 +0x75
net/rpc/jsonrpc.(*serverCodec).ReadRequestHeader(0xc000d06060, 0xc00476d080)
	/usr/local/go/src/net/rpc/jsonrpc/server.go:66 +0x6c
net/rpc.(*Server).readRequestHeader(0xc000c820f0, {0x20c43c8, 0xc000d06060})
	/usr/local/go/src/net/rpc/server.go:587 +0x58
net/rpc.(*Server).readRequest(0xc000c820f0?, {0x20c43c8, 0xc000d06060})
	/usr/local/go/src/net/rpc/server.go:547 +0x2f
net/rpc.(*Server).ServeCodec(0xc000c820f0, {0x20c43c8, 0xc000d06060})
	/usr/local/go/src/net/rpc/server.go:462 +0x71
created by github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener.func1 in goroutine 1607
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:214 +0x2b2

goroutine 1614 [select, 9 minutes]:
github.com/google/syzkaller/vm/proxyapp.ctor.func1()
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:59 +0x13c
created by github.com/google/syzkaller/vm/proxyapp.ctor in goroutine 1606
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:47 +0x1c5

goroutine 13640537 [runnable]:
github.com/stretchr/testify/assert.CallerInfo()
	/syzkaller/gopath/pkg/mod/github.com/stretchr/testify@v1.11.1/assert/assertions.go:264 +0x4ae
github.com/stretchr/testify/mock.(*Mock).MethodCalled(0xc000be0500, {0x24a16a0, 0x8}, {0xc25d000e60, 0x2, 0x2})
	/syzkaller/gopath/pkg/mod/github.com/stretchr/testify@v1.11.1/mock/mock.go:558 +0xa18
github.com/stretchr/testify/mock.(*Mock).Called(0xc000be0500, {0xc25d000e60, 0x2, 0x2})
	/syzkaller/gopath/pkg/mod/github.com/stretchr/testify@v1.11.1/mock/mock.go:491 +0x125
github.com/google/syzkaller/vm/proxyapp/mocks.(*ProxyAppInterface).PoolLogs(0xc000be0500, {}, 0xc24c3f67b0)
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/mocks/ProxyAppInterface.go:383 +0x85
reflect.Value.call({0xc000b1b0e0?, 0xc000b32558?, 0x13?}, {0x15cfc8b, 0x4}, {0xc000ae6ef8, 0x3, 0x3?})
	/usr/local/go/src/reflect/value.go:584 +0xca6
reflect.Value.Call({0xc000b1b0e0?, 0xc000b32558?, 0x0?}, {0xc05dc8aef8?, 0x0?, 0x0?})
	/usr/local/go/src/reflect/value.go:368 +0xb9
net/rpc.(*service).call(0xc000b276c0, 0xc000be0550, 0xc0005562a0, 0xc0005562b0, 0xc000be6500, 0xc000538640, {0x1318520?, 0x2e4a8c0?, 0x0?}, {0x1233f60, ...}, ...)
	/usr/local/go/src/net/rpc/server.go:381 +0x209
created by net/rpc.(*Server).ServeCodec in goroutine 108
	/usr/local/go/src/net/rpc/server.go:478 +0x3ca

goroutine 13640561 [runnable]:
github.com/stretchr/testify/assert.CallerInfo()
	/syzkaller/gopath/pkg/mod/github.com/stretchr/testify@v1.11.1/assert/assertions.go:264 +0x48a
github.com/stretchr/testify/mock.(*Mock).MethodCalled(0xc000c820a0, {0x24a16a0, 0x8}, {0xc25d557180, 0x2, 0x2})
	/syzkaller/gopath/pkg/mod/github.com/stretchr/testify@v1.11.1/mock/mock.go:558 +0xa18
github.com/stretchr/testify/mock.(*Mock).Called(0xc000c820a0, {0xc25d557180, 0x2, 0x2})
	/syzkaller/gopath/pkg/mod/github.com/stretchr/testify@v1.11.1/mock/mock.go:491 +0x125
github.com/google/syzkaller/vm/proxyapp/mocks.(*ProxyAppInterface).PoolLogs(0xc000c820a0, {}, 0xc24c52fd58)
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/mocks/ProxyAppInterface.go:383 +0x85
reflect.Value.call({0xc000b1b0e0?, 0xc00011e220?, 0x13?}, {0x15cfc8b, 0x4}, {0xc0008b6ef8, 0x3, 0x3?})
	/usr/local/go/src/reflect/value.go:584 +0xca6
reflect.Value.Call({0xc000b1b0e0?, 0xc00011e220?, 0xc000812300?}, {0xc0008996f8?, 0x8f70be?, 0xc000812300?})
	/usr/local/go/src/reflect/value.go:368 +0xb9
net/rpc.(*service).call(0xc000c9a040, 0xc000c820f0, 0xc0001e7798, 0xc0001e77a0, 0xc000d86580, 0xc000a31a20, {0x1318520?, 0x2e4a8c0?, 0x15e7c4b?}, {0x1233f60, ...}, ...)
	/usr/local/go/src/net/rpc/server.go:381 +0x209
created by net/rpc.(*Server).ServeCodec in goroutine 455
	/usr/local/go/src/net/rpc/server.go:478 +0x3ca

goroutine 13640628 [runnable]:
fmt.(*pp).printArg(0xc04f7080d0?, {0x1275280?, 0xc19da76530?}, 0x64?)
	/usr/local/go/src/fmt/print.go:681 +0x5af
fmt.(*pp).doPrintf(0xc04f7080d0, {0x15d0bf3, 0x5}, {0xc000f9b440, 0x2, 0x2})
	/usr/local/go/src/fmt/print.go:1074 +0x37e
fmt.Sprintf({0x15d0bf3, 0x5}, {0xc000f9b440, 0x2, 0x2})
	/usr/local/go/src/fmt/print.go:239 +0x53
github.com/stretchr/testify/assert.CallerInfo()
	/syzkaller/gopath/pkg/mod/github.com/stretchr/testify@v1.11.1/assert/assertions.go:264 +0x4e6
github.com/stretchr/testify/mock.(*Mock).MethodCalled(0xc0009980a0, {0x24a16a0, 0x8}, {0xc25d5ac4a0, 0x2, 0x2})
	/syzkaller/gopath/pkg/mod/github.com/stretchr/testify@v1.11.1/mock/mock.go:558 +0xa18
github.com/stretchr/testify/mock.(*Mock).Called(0xc0009980a0, {0xc25d5ac4a0, 0x2, 0x2})
	/syzkaller/gopath/pkg/mod/github.com/stretchr/testify@v1.11.1/mock/mock.go:491 +0x125
github.com/google/syzkaller/vm/proxyapp/mocks.(*ProxyAppInterface).PoolLogs(0xc0009980a0, {}, 0xc24c373248)
	/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/mocks/ProxyAppInterface.go:383 +0x85
reflect.Value.call({0xc000b1b0e0?, 0xc0008bc050?, 0x13?}, {0x15cfc8b, 0x4}, {0xc032e08ef8, 0x3, 0x3?})
	/usr/local/go/src/reflect/value.go:584 +0xca6
reflect.Value.Call({0xc000b1b0e0?, 0xc0008bc050?, 0x702a2820203a5353?}, {0xc000c526f8?, 0x73676f4c6c6f6f50?, 0x7b263d796c706552?})
	/usr/local/go/src/reflect/value.go:368 +0xb9
net/rpc.(*service).call(0xc0009b8080, 0xc0009980f0, 0xc000b10380, 0xc000b10390, 0xc000c58600, 0xc000d97900, {0x1318520?, 0x2e4a8c0?, 0x3d676e6972747328?}, {0x1233f60, ...}, ...)
	/usr/local/go/src/net/rpc/server.go:381 +0x209
created by net/rpc.(*Server).ServeCodec in goroutine 1511
	/usr/local/go/src/net/rpc/server.go:478 +0x3ca
FAIL	github.com/google/syzkaller/vm/proxyapp	601.324s
```

Observed again on  61210f3f83c6a99132eb7e851a85d2554d2b2c22
https://github.com/google/syzkaller/actions/runs/24338203040/job/71060282541

```
2026-04-13T10:36:35.7967372Z 2026/04/13 10:26:35 error pooling ProxyApp logs: connection is shut down
2026-04-13T10:36:35.7967697Z coverage: 38.0% of statements
2026-04-13T10:36:35.7967927Z panic: test timed out after 10m0s
2026-04-13T10:36:35.7968142Z 	running tests:
2026-04-13T10:36:35.7968347Z 		TestCtor_TCP_Reconnect_PoolChanged (10m0s)
2026-04-13T10:36:35.7968529Z 
2026-04-13T10:36:35.7968611Z goroutine 23748953 [running]:
2026-04-13T10:36:35.7968825Z testing.(*M).startAlarm.func1()
2026-04-13T10:36:35.7969078Z 	/usr/local/go/src/testing/testing.go:2802 +0x34b
2026-04-13T10:36:35.7969336Z created by time.goFunc
2026-04-13T10:36:35.7969710Z 	/usr/local/go/src/time/sleep.go:215 +0x2d
2026-04-13T10:36:35.7969881Z 
2026-04-13T10:36:35.7969969Z goroutine 1 [chan receive, 9 minutes]:
2026-04-13T10:36:35.7970326Z testing.(*T).Run(0x394e229ea008, {0x1633804?, 0x394e2278fb30?}, 0x201b4e0)
2026-04-13T10:36:35.7970696Z 	/usr/local/go/src/testing/testing.go:2109 +0x4e5
2026-04-13T10:36:35.7970969Z testing.runTests.func1(0x394e229ea008)
2026-04-13T10:36:35.7971227Z 	/usr/local/go/src/testing/testing.go:2585 +0x37
2026-04-13T10:36:35.7971511Z testing.tRunner(0x394e229ea008, 0x394e2278fc58)
2026-04-13T10:36:35.7971789Z 	/usr/local/go/src/testing/testing.go:2036 +0xea
2026-04-13T10:36:35.7972327Z testing.runTests({0x1625538, 0x1b}, {0x163dc70, 0x27}, 0x394e229a21c8, {0x2f624e0, 0x1f, 0x1f}, {0xc26f4facd9250555, 0x8bdb08fbac, ...})
2026-04-13T10:36:35.7972885Z 	/usr/local/go/src/testing/testing.go:2583 +0x505
2026-04-13T10:36:35.7973147Z testing.(*M).Run(0x394e229d3040)
2026-04-13T10:36:35.7973386Z 	/usr/local/go/src/testing/testing.go:2443 +0x6ac
2026-04-13T10:36:35.7973635Z main.main()
2026-04-13T10:36:35.7973793Z 	_testmain.go:116 +0x9b
2026-04-13T10:36:35.7973917Z 
2026-04-13T10:36:35.7973991Z goroutine 28 [chan receive]:
2026-04-13T10:36:35.7974246Z github.com/google/syzkaller/pkg/stat.newSet.func1()
2026-04-13T10:36:35.7974688Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/pkg/stat/set.go:95 +0x66
2026-04-13T10:36:35.7975190Z created by github.com/google/syzkaller/pkg/stat.newSet in goroutine 1
2026-04-13T10:36:35.7975671Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/pkg/stat/set.go:94 +0xd6
2026-04-13T10:36:35.7976005Z 
2026-04-13T10:36:35.7976092Z goroutine 1252 [chan receive, 9 minutes]:
2026-04-13T10:36:35.7976504Z github.com/google/syzkaller/vm/proxyapp.TestCtor_TCP_Reconnect_PoolChanged(0x394e22978488)
2026-04-13T10:36:35.7977299Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:188 +0x3df
2026-04-13T10:36:35.7977821Z testing.tRunner(0x394e22978488, 0x201b4e0)
2026-04-13T10:36:35.7978093Z 	/usr/local/go/src/testing/testing.go:2036 +0xea
2026-04-13T10:36:35.7978360Z created by testing.(*T).Run in goroutine 1
2026-04-13T10:36:35.7978736Z 	/usr/local/go/src/testing/testing.go:2101 +0x4c5
2026-04-13T10:36:35.7978924Z 
2026-04-13T10:36:35.7979007Z goroutine 1235 [IO wait, 9 minutes]:
2026-04-13T10:36:35.7979273Z internal/poll.runtime_pollWait(0x7f4040c23e00, 0x72)
2026-04-13T10:36:35.7979940Z 	/usr/local/go/src/runtime/netpoll.go:351 +0x85
2026-04-13T10:36:35.7980289Z internal/poll.(*pollDesc).wait(0x394e22e0c280?, 0x7f404084f468?, 0x0)
2026-04-13T10:36:35.7980688Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
2026-04-13T10:36:35.7980999Z internal/poll.(*pollDesc).waitRead(...)
2026-04-13T10:36:35.7981276Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
2026-04-13T10:36:35.7981574Z internal/poll.(*FD).Accept(0x394e22e0c280)
2026-04-13T10:36:35.7981849Z 	/usr/local/go/src/internal/poll/fd_unix.go:613 +0x28c
2026-04-13T10:36:35.7982123Z net.(*netFD).accept(0x394e22e0c280)
2026-04-13T10:36:35.7982362Z 	/usr/local/go/src/net/fd_unix.go:150 +0x29
2026-04-13T10:36:35.7982612Z net.(*TCPListener).accept(0x394e22c84b40)
2026-04-13T10:36:35.7982886Z 	/usr/local/go/src/net/tcpsock_posix.go:159 +0x1b
2026-04-13T10:36:35.7983155Z net.(*TCPListener).Accept(0x394e22c84b40)
2026-04-13T10:36:35.7983401Z 	/usr/local/go/src/net/tcpsock.go:387 +0x30
2026-04-13T10:36:35.7983845Z github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener.func1()
2026-04-13T10:36:35.7984504Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:210 +0x5f
2026-04-13T10:36:35.7985234Z created by github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener in goroutine 1234
2026-04-13T10:36:35.7985961Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:208 +0x291
2026-04-13T10:36:35.7986381Z 
2026-04-13T10:36:35.7986466Z goroutine 281 [select, 9 minutes]:
2026-04-13T10:36:35.7986737Z github.com/google/syzkaller/vm/proxyapp.ctor.func1()
2026-04-13T10:36:35.7987236Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:59 +0x165
2026-04-13T10:36:35.7987801Z created by github.com/google/syzkaller/vm/proxyapp.ctor in goroutine 108
2026-04-13T10:36:35.7988350Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:47 +0x27b
2026-04-13T10:36:35.7988732Z 
2026-04-13T10:36:35.7988809Z goroutine 1273 [IO wait]:
2026-04-13T10:36:35.7989046Z internal/poll.runtime_pollWait(0x7f4040c23400, 0x72)
2026-04-13T10:36:35.7989348Z 	/usr/local/go/src/runtime/netpoll.go:351 +0x85
2026-04-13T10:36:35.7989822Z internal/poll.(*pollDesc).wait(0x394e22ea3100?, 0x394e22dc6601?, 0x0)
2026-04-13T10:36:35.7990209Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
2026-04-13T10:36:35.7990521Z internal/poll.(*pollDesc).waitRead(...)
2026-04-13T10:36:35.7990808Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
2026-04-13T10:36:35.7991163Z internal/poll.(*FD).Read(0x394e22ea3100, {0x394e22dc6601, 0x5ff, 0x5ff})
2026-04-13T10:36:35.7991522Z 	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
2026-04-13T10:36:35.7991936Z net.(*netFD).Read(0x394e22ea3100, {0x394e22dc6601?, 0x41dfff?, 0x394e22a67d48?})
2026-04-13T10:36:35.7992314Z 	/usr/local/go/src/net/fd_posix.go:68 +0x25
2026-04-13T10:36:35.7992652Z net.(*conn).Read(0x394e22182050, {0x394e22dc6601?, 0x41ddb4?, 0x39522ee79618?})
2026-04-13T10:36:35.7992991Z 	/usr/local/go/src/net/net.go:196 +0x45
2026-04-13T10:36:35.7993252Z encoding/json.(*Decoder).refill(0x394e22485680)
2026-04-13T10:36:35.7993539Z 	/usr/local/go/src/encoding/json/stream.go:167 +0x188
2026-04-13T10:36:35.7993843Z encoding/json.(*Decoder).readValue(0x394e22485680)
2026-04-13T10:36:35.7994143Z 	/usr/local/go/src/encoding/json/stream.go:142 +0x85
2026-04-13T10:36:35.7994648Z encoding/json.(*Decoder).Decode(0x394e22485680, {0x1308de0, 0x394e22bf6ac8})
2026-04-13T10:36:35.7995030Z 	/usr/local/go/src/encoding/json/stream.go:65 +0x75
2026-04-13T10:36:35.7995427Z net/rpc/jsonrpc.(*clientCodec).ReadResponseHeader(0x394e22bf6a80, 0x394e22bcd620)
2026-04-13T10:36:35.7995942Z 	/usr/local/go/src/net/rpc/jsonrpc/client.go:76 +0x65
2026-04-13T10:36:35.7996220Z net/rpc.(*Client).input(0x394e22f84060)
2026-04-13T10:36:35.7996474Z 	/usr/local/go/src/net/rpc/client.go:109 +0x9f
2026-04-13T10:36:35.7996782Z created by net/rpc.NewClientWithCodec in goroutine 1252
2026-04-13T10:36:35.7997091Z 	/usr/local/go/src/net/rpc/client.go:206 +0xb6
2026-04-13T10:36:35.7997266Z 
2026-04-13T10:36:35.7997343Z goroutine 1274 [runnable]:
2026-04-13T10:36:35.7997658Z github.com/google/syzkaller/vm/proxyapp.(*ProxyApp).doLogPooling.func1()
2026-04-13T10:36:35.7998228Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:324 +0x13b
2026-04-13T10:36:35.7998868Z created by github.com/google/syzkaller/vm/proxyapp.(*ProxyApp).doLogPooling in goroutine 1252
2026-04-13T10:36:35.7999792Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:314 +0x110
2026-04-13T10:36:35.8000193Z 
2026-04-13T10:36:35.8000268Z goroutine 104 [select]:
2026-04-13T10:36:35.8000595Z github.com/google/syzkaller/vm/proxyapp.(*ProxyApp).doLogPooling.func1()
2026-04-13T10:36:35.8001232Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:324 +0x13b
2026-04-13T10:36:35.8001880Z created by github.com/google/syzkaller/vm/proxyapp.(*ProxyApp).doLogPooling in goroutine 124
2026-04-13T10:36:35.8002506Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:314 +0x110
2026-04-13T10:36:35.8002900Z 
2026-04-13T10:36:35.8002972Z goroutine 1282 [IO wait]:
2026-04-13T10:36:35.8003213Z internal/poll.runtime_pollWait(0x7f4040c23800, 0x72)
2026-04-13T10:36:35.8003516Z 	/usr/local/go/src/runtime/netpoll.go:351 +0x85
2026-04-13T10:36:35.8003859Z internal/poll.(*pollDesc).wait(0x394e22f00000?, 0x394e22f1a001?, 0x0)
2026-04-13T10:36:35.8004235Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
2026-04-13T10:36:35.8004547Z internal/poll.(*pollDesc).waitRead(...)
2026-04-13T10:36:35.8004835Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
2026-04-13T10:36:35.8005185Z internal/poll.(*FD).Read(0x394e22f00000, {0x394e22f1a001, 0x5ff, 0x5ff})
2026-04-13T10:36:35.8005546Z 	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
2026-04-13T10:36:35.8005903Z net.(*netFD).Read(0x394e22f00000, {0x394e22f1a001?, 0x550c6e?, 0x12bbb20?})
2026-04-13T10:36:35.8006241Z 	/usr/local/go/src/net/fd_posix.go:68 +0x25
2026-04-13T10:36:35.8006537Z net.(*conn).Read(0x394e22766000, {0x394e22f1a001?, 0x1?, 0x1?})
2026-04-13T10:36:35.8006831Z 	/usr/local/go/src/net/net.go:196 +0x45
2026-04-13T10:36:35.8007093Z encoding/json.(*Decoder).refill(0x394e22596780)
2026-04-13T10:36:35.8007391Z 	/usr/local/go/src/encoding/json/stream.go:167 +0x188
2026-04-13T10:36:35.8007691Z encoding/json.(*Decoder).readValue(0x394e22596780)
2026-04-13T10:36:35.8007993Z 	/usr/local/go/src/encoding/json/stream.go:142 +0x85
2026-04-13T10:36:35.8008363Z encoding/json.(*Decoder).Decode(0x394e22596780, {0x1308d80, 0x394e22f04080})
2026-04-13T10:36:35.8008736Z 	/usr/local/go/src/encoding/json/stream.go:65 +0x75
2026-04-13T10:36:35.8009122Z net/rpc/jsonrpc.(*serverCodec).ReadRequestHeader(0x394e22f04060, 0x394e22e80200)
2026-04-13T10:36:35.8009649Z 	/usr/local/go/src/net/rpc/jsonrpc/server.go:66 +0x6d
2026-04-13T10:36:35.8010040Z net/rpc.(*Server).readRequestHeader(0x394e22c33e50, {0x2058cb8, 0x394e22f04060})
2026-04-13T10:36:35.8010430Z 	/usr/local/go/src/net/rpc/server.go:589 +0x58
2026-04-13T10:36:35.8010770Z net/rpc.(*Server).readRequest(0x394e22c33e50?, {0x2058cb8, 0x394e22f04060})
2026-04-13T10:36:35.8021201Z 	/usr/local/go/src/net/rpc/server.go:549 +0x2f
2026-04-13T10:36:35.8021844Z net/rpc.(*Server).ServeCodec(0x394e22c33e50, {0x2058cb8, 0x394e22f04060})
2026-04-13T10:36:35.8022229Z 	/usr/local/go/src/net/rpc/server.go:464 +0x71
2026-04-13T10:36:35.8022777Z created by github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener.func1 in goroutine 1235
2026-04-13T10:36:35.8023556Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:214 +0x2b2
2026-04-13T10:36:35.8024107Z 
2026-04-13T10:36:35.8024193Z goroutine 1251 [select, 9 minutes]:
2026-04-13T10:36:35.8024480Z github.com/google/syzkaller/vm/proxyapp.ctor.func1()
2026-04-13T10:36:35.8024991Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:59 +0x165
2026-04-13T10:36:35.8025587Z created by github.com/google/syzkaller/vm/proxyapp.ctor in goroutine 1234
2026-04-13T10:36:35.8026157Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:47 +0x27b
2026-04-13T10:36:35.8026545Z 
2026-04-13T10:36:35.8026621Z goroutine 1151 [IO wait]:
2026-04-13T10:36:35.8026876Z internal/poll.runtime_pollWait(0x7f4040991200, 0x72)
2026-04-13T10:36:35.8027186Z 	/usr/local/go/src/runtime/netpoll.go:351 +0x85
2026-04-13T10:36:35.8027534Z internal/poll.(*pollDesc).wait(0x394e22d42780?, 0x394e22e06a80?, 0x0)
2026-04-13T10:36:35.8027925Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
2026-04-13T10:36:35.8028247Z internal/poll.(*pollDesc).waitRead(...)
2026-04-13T10:36:35.8028533Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
2026-04-13T10:36:35.8028899Z internal/poll.(*FD).Read(0x394e22d42780, {0x394e22e06a80, 0xa80, 0xa80})
2026-04-13T10:36:35.8029258Z 	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
2026-04-13T10:36:35.8029835Z net.(*netFD).Read(0x394e22d42780, {0x394e22e06a80?, 0x394e22e06a80?, 0x5?})
2026-04-13T10:36:35.8030187Z 	/usr/local/go/src/net/fd_posix.go:68 +0x25
2026-04-13T10:36:35.8030541Z net.(*conn).Read(0x394e22602008, {0x394e22e06a80?, 0x7f3fc0072f08?, 0x7f40888828f8?})
2026-04-13T10:36:35.8030939Z 	/usr/local/go/src/net/net.go:196 +0x45
2026-04-13T10:36:35.8031363Z crypto/tls.(*atLeastReader).Read(0x39523f580168, {0x394e22e06a80?, 0x394e227f85a0?, 0x394e75badb20?})
2026-04-13T10:36:35.8031817Z 	/usr/local/go/src/crypto/tls/conn.go:815 +0x3b
2026-04-13T10:36:35.8032155Z bytes.(*Buffer).ReadFrom(0x394e22613428, {0x20313c0, 0x39523f580168})
2026-04-13T10:36:35.8032494Z 	/usr/local/go/src/bytes/buffer.go:229 +0x98
2026-04-13T10:36:35.8032905Z crypto/tls.(*Conn).readFromUntil(0x394e22613188, {0x2031680, 0x394e22602008}, 0x394e75badce0?)
2026-04-13T10:36:35.8033333Z 	/usr/local/go/src/crypto/tls/conn.go:837 +0xde
2026-04-13T10:36:35.8033637Z crypto/tls.(*Conn).readRecordOrCCS(0x394e22613188, 0x0)
2026-04-13T10:36:35.8033947Z 	/usr/local/go/src/crypto/tls/conn.go:626 +0x3db
2026-04-13T10:36:35.8034220Z crypto/tls.(*Conn).readRecord(...)
2026-04-13T10:36:35.8034460Z 	/usr/local/go/src/crypto/tls/conn.go:588
2026-04-13T10:36:35.8034810Z crypto/tls.(*Conn).Read(0x394e22613188, {0x394e22934001, 0x5ff, 0x39522dd56bf8?})
2026-04-13T10:36:35.8035188Z 	/usr/local/go/src/crypto/tls/conn.go:1393 +0x145
2026-04-13T10:36:35.8035477Z encoding/json.(*Decoder).refill(0x394e226203c0)
2026-04-13T10:36:35.8035780Z 	/usr/local/go/src/encoding/json/stream.go:167 +0x188
2026-04-13T10:36:35.8036082Z encoding/json.(*Decoder).readValue(0x394e226203c0)
2026-04-13T10:36:35.8036393Z 	/usr/local/go/src/encoding/json/stream.go:142 +0x85
2026-04-13T10:36:35.8036759Z encoding/json.(*Decoder).Decode(0x394e226203c0, {0x1308de0, 0x394e22e0c0c8})
2026-04-13T10:36:35.8037131Z 	/usr/local/go/src/encoding/json/stream.go:65 +0x75
2026-04-13T10:36:35.8037528Z net/rpc/jsonrpc.(*clientCodec).ReadResponseHeader(0x394e22e0c080, 0x394e22303cb0)
2026-04-13T10:36:35.8037933Z 	/usr/local/go/src/net/rpc/jsonrpc/client.go:76 +0x65
2026-04-13T10:36:35.8038214Z net/rpc.(*Client).input(0x394e229b4d20)
2026-04-13T10:36:35.8038482Z 	/usr/local/go/src/net/rpc/client.go:109 +0x9f
2026-04-13T10:36:35.8038787Z created by net/rpc.NewClientWithCodec in goroutine 108
2026-04-13T10:36:35.8039216Z 	/usr/local/go/src/net/rpc/client.go:206 +0xb6
2026-04-13T10:36:35.8039398Z 
2026-04-13T10:36:35.8039774Z goroutine 125 [IO wait, 10 minutes]:
2026-04-13T10:36:35.8040060Z internal/poll.runtime_pollWait(0x7f4040991800, 0x72)
2026-04-13T10:36:35.8040353Z 	/usr/local/go/src/runtime/netpoll.go:351 +0x85
2026-04-13T10:36:35.8040820Z internal/poll.(*pollDesc).wait(0x394e228e0380?, 0x900000036?, 0x0)
2026-04-13T10:36:35.8041200Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
2026-04-13T10:36:35.8041517Z internal/poll.(*pollDesc).waitRead(...)
2026-04-13T10:36:35.8041796Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
2026-04-13T10:36:35.8042087Z internal/poll.(*FD).Accept(0x394e228e0380)
2026-04-13T10:36:35.8042372Z 	/usr/local/go/src/internal/poll/fd_unix.go:613 +0x28c
2026-04-13T10:36:35.8042643Z net.(*netFD).accept(0x394e228e0380)
2026-04-13T10:36:35.8042879Z 	/usr/local/go/src/net/fd_unix.go:150 +0x29
2026-04-13T10:36:35.8043133Z net.(*TCPListener).accept(0x394e225d6280)
2026-04-13T10:36:35.8043417Z 	/usr/local/go/src/net/tcpsock_posix.go:159 +0x1b
2026-04-13T10:36:35.8043695Z net.(*TCPListener).Accept(0x394e225d6280)
2026-04-13T10:36:35.8043944Z 	/usr/local/go/src/net/tcpsock.go:387 +0x30
2026-04-13T10:36:35.8044360Z github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener.func1()
2026-04-13T10:36:35.8045038Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:210 +0x5f
2026-04-13T10:36:35.8045782Z created by github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener in goroutine 124
2026-04-13T10:36:35.8046524Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:208 +0x291
2026-04-13T10:36:35.8046953Z 
2026-04-13T10:36:35.8047032Z goroutine 103 [IO wait]:
2026-04-13T10:36:35.8047276Z internal/poll.runtime_pollWait(0x7f4040991000, 0x72)
2026-04-13T10:36:35.8047586Z 	/usr/local/go/src/runtime/netpoll.go:351 +0x85
2026-04-13T10:36:35.8047930Z internal/poll.(*pollDesc).wait(0x394e228e0a80?, 0x394e22966601?, 0x0)
2026-04-13T10:36:35.8048323Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
2026-04-13T10:36:35.8048640Z internal/poll.(*pollDesc).waitRead(...)
2026-04-13T10:36:35.8048922Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
2026-04-13T10:36:35.8049287Z internal/poll.(*FD).Read(0x394e228e0a80, {0x394e22966601, 0x5ff, 0x5ff})
2026-04-13T10:36:35.8049813Z 	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
2026-04-13T10:36:35.8050184Z net.(*netFD).Read(0x394e228e0a80, {0x394e22966601?, 0x41dfff?, 0x394e2366fd48?})
2026-04-13T10:36:35.8050550Z 	/usr/local/go/src/net/fd_posix.go:68 +0x25
2026-04-13T10:36:35.8050885Z net.(*conn).Read(0x394e22900150, {0x394e22966601?, 0x41ddb4?, 0x39522f0a40b8?})
2026-04-13T10:36:35.8051261Z 	/usr/local/go/src/net/net.go:196 +0x45
2026-04-13T10:36:35.8051521Z encoding/json.(*Decoder).refill(0x394e2294e000)
2026-04-13T10:36:35.8051814Z 	/usr/local/go/src/encoding/json/stream.go:167 +0x188
2026-04-13T10:36:35.8052128Z encoding/json.(*Decoder).readValue(0x394e2294e000)
2026-04-13T10:36:35.8052440Z 	/usr/local/go/src/encoding/json/stream.go:142 +0x85
2026-04-13T10:36:35.8052801Z encoding/json.(*Decoder).Decode(0x394e2294e000, {0x1308de0, 0x394e2293a248})
2026-04-13T10:36:35.8053178Z 	/usr/local/go/src/encoding/json/stream.go:65 +0x75
2026-04-13T10:36:35.8053574Z net/rpc/jsonrpc.(*clientCodec).ReadResponseHeader(0x394e2293a200, 0x394e228d56e0)
2026-04-13T10:36:35.8053985Z 	/usr/local/go/src/net/rpc/jsonrpc/client.go:76 +0x65
2026-04-13T10:36:35.8054268Z net/rpc.(*Client).input(0x394e2292e960)
2026-04-13T10:36:35.8054519Z 	/usr/local/go/src/net/rpc/client.go:109 +0x9f
2026-04-13T10:36:35.8054822Z created by net/rpc.NewClientWithCodec in goroutine 124
2026-04-13T10:36:35.8055131Z 	/usr/local/go/src/net/rpc/client.go:206 +0xb6
2026-04-13T10:36:35.8055310Z 
2026-04-13T10:36:35.8055386Z goroutine 128 [runnable]:
2026-04-13T10:36:35.8055628Z internal/poll.runtime_pollWait(0x7f4040991400, 0x72)
2026-04-13T10:36:35.8056064Z 	/usr/local/go/src/runtime/netpoll.go:351 +0x85
2026-04-13T10:36:35.8056406Z internal/poll.(*pollDesc).wait(0x394e228e0b00?, 0x394e22966001?, 0x0)
2026-04-13T10:36:35.8056792Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
2026-04-13T10:36:35.8057097Z internal/poll.(*pollDesc).waitRead(...)
2026-04-13T10:36:35.8057488Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
2026-04-13T10:36:35.8057848Z internal/poll.(*FD).Read(0x394e228e0b00, {0x394e22966001, 0x5ff, 0x5ff})
2026-04-13T10:36:35.8058204Z 	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
2026-04-13T10:36:35.8058569Z net.(*netFD).Read(0x394e228e0b00, {0x394e22966001?, 0x550c6e?, 0x12bbb20?})
2026-04-13T10:36:35.8058916Z 	/usr/local/go/src/net/fd_posix.go:68 +0x25
2026-04-13T10:36:35.8059208Z net.(*conn).Read(0x394e2289a200, {0x394e22966001?, 0x1?, 0x1?})
2026-04-13T10:36:35.8059793Z 	/usr/local/go/src/net/net.go:196 +0x45
2026-04-13T10:36:35.8060056Z encoding/json.(*Decoder).refill(0x394e228e5400)
2026-04-13T10:36:35.8060362Z 	/usr/local/go/src/encoding/json/stream.go:167 +0x188
2026-04-13T10:36:35.8060670Z encoding/json.(*Decoder).readValue(0x394e228e5400)
2026-04-13T10:36:35.8060965Z 	/usr/local/go/src/encoding/json/stream.go:142 +0x85
2026-04-13T10:36:35.8061442Z encoding/json.(*Decoder).Decode(0x394e228e5400, {0x1308d80, 0x394e228b4aa0})
2026-04-13T10:36:35.8061825Z 	/usr/local/go/src/encoding/json/stream.go:65 +0x75
2026-04-13T10:36:35.8062212Z net/rpc/jsonrpc.(*serverCodec).ReadRequestHeader(0x394e228b4a80, 0x394e228aa900)
2026-04-13T10:36:35.8062623Z 	/usr/local/go/src/net/rpc/jsonrpc/server.go:66 +0x6d
2026-04-13T10:36:35.8063002Z net/rpc.(*Server).readRequestHeader(0x394e228cc550, {0x2058cb8, 0x394e228b4a80})
2026-04-13T10:36:35.8063381Z 	/usr/local/go/src/net/rpc/server.go:589 +0x58
2026-04-13T10:36:35.8063720Z net/rpc.(*Server).readRequest(0x394e228cc550?, {0x2058cb8, 0x394e228b4a80})
2026-04-13T10:36:35.8064069Z 	/usr/local/go/src/net/rpc/server.go:549 +0x2f
2026-04-13T10:36:35.8064411Z net/rpc.(*Server).ServeCodec(0x394e228cc550, {0x2058cb8, 0x394e228b4a80})
2026-04-13T10:36:35.8064762Z 	/usr/local/go/src/net/rpc/server.go:464 +0x71
2026-04-13T10:36:35.8065265Z created by github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener.func1 in goroutine 125
2026-04-13T10:36:35.8066029Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:214 +0x2b2
2026-04-13T10:36:35.8066457Z 
2026-04-13T10:36:35.8066545Z goroutine 107 [select, 10 minutes]:
2026-04-13T10:36:35.8066821Z github.com/google/syzkaller/vm/proxyapp.ctor.func1()
2026-04-13T10:36:35.8067319Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:59 +0x165
2026-04-13T10:36:35.8067873Z created by github.com/google/syzkaller/vm/proxyapp.ctor in goroutine 124
2026-04-13T10:36:35.8068424Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:47 +0x27b
2026-04-13T10:36:35.8068815Z 
2026-04-13T10:36:35.8068893Z goroutine 230 [IO wait, 9 minutes]:
2026-04-13T10:36:35.8069151Z internal/poll.runtime_pollWait(0x7f4040c23200, 0x72)
2026-04-13T10:36:35.8069578Z 	/usr/local/go/src/runtime/netpoll.go:351 +0x85
2026-04-13T10:36:35.8069916Z internal/poll.(*pollDesc).wait(0x394e22d42080?, 0x900427137?, 0x0)
2026-04-13T10:36:35.8070296Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
2026-04-13T10:36:35.8070607Z internal/poll.(*pollDesc).waitRead(...)
2026-04-13T10:36:35.8070889Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
2026-04-13T10:36:35.8071176Z internal/poll.(*FD).Accept(0x394e22d42080)
2026-04-13T10:36:35.8071456Z 	/usr/local/go/src/internal/poll/fd_unix.go:613 +0x28c
2026-04-13T10:36:35.8071719Z net.(*netFD).accept(0x394e22d42080)
2026-04-13T10:36:35.8071952Z 	/usr/local/go/src/net/fd_unix.go:150 +0x29
2026-04-13T10:36:35.8072208Z net.(*TCPListener).accept(0x394e22d3e040)
2026-04-13T10:36:35.8072475Z 	/usr/local/go/src/net/tcpsock_posix.go:159 +0x1b
2026-04-13T10:36:35.8072750Z net.(*TCPListener).Accept(0x394e22d3e040)
2026-04-13T10:36:35.8073152Z 	/usr/local/go/src/net/tcpsock.go:387 +0x30
2026-04-13T10:36:35.8073430Z crypto/tls.(*listener).Accept(0x394e22d32168)
2026-04-13T10:36:35.8073698Z 	/usr/local/go/src/crypto/tls/tls.go:78 +0x24
2026-04-13T10:36:35.8074112Z github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener.func1()
2026-04-13T10:36:35.8074906Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:210 +0x5f
2026-04-13T10:36:35.8075630Z created by github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener in goroutine 108
2026-04-13T10:36:35.8076347Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:208 +0x291
2026-04-13T10:36:35.8076770Z 
2026-04-13T10:36:35.8076848Z goroutine 1299 [select, 9 minutes]:
2026-04-13T10:36:35.8077120Z github.com/google/syzkaller/vm/proxyapp.ctor.func1()
2026-04-13T10:36:35.8077620Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:59 +0x165
2026-04-13T10:36:35.8078182Z created by github.com/google/syzkaller/vm/proxyapp.ctor in goroutine 1252
2026-04-13T10:36:35.8078733Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:47 +0x27b
2026-04-13T10:36:35.8079128Z 
2026-04-13T10:36:35.8079200Z goroutine 1052 [select]:
2026-04-13T10:36:35.8079743Z github.com/google/syzkaller/vm/proxyapp.(*ProxyApp).doLogPooling.func1()
2026-04-13T10:36:35.8080324Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:324 +0x13b
2026-04-13T10:36:35.8080960Z created by github.com/google/syzkaller/vm/proxyapp.(*ProxyApp).doLogPooling in goroutine 1251
2026-04-13T10:36:35.8081596Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:314 +0x110
2026-04-13T10:36:35.8081982Z 
2026-04-13T10:36:35.8082054Z goroutine 1051 [IO wait]:
2026-04-13T10:36:35.8082298Z internal/poll.runtime_pollWait(0x7f4040c23a00, 0x72)
2026-04-13T10:36:35.8082605Z 	/usr/local/go/src/runtime/netpoll.go:351 +0x85
2026-04-13T10:36:35.8082946Z internal/poll.(*pollDesc).wait(0x394e22dc8080?, 0x394e22bfe001?, 0x0)
2026-04-13T10:36:35.8083335Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
2026-04-13T10:36:35.8083672Z internal/poll.(*pollDesc).waitRead(...)
2026-04-13T10:36:35.8083953Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
2026-04-13T10:36:35.8084311Z internal/poll.(*FD).Read(0x394e22dc8080, {0x394e22bfe001, 0x5ff, 0x5ff})
2026-04-13T10:36:35.8084667Z 	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
2026-04-13T10:36:35.8085047Z net.(*netFD).Read(0x394e22dc8080, {0x394e22bfe001?, 0x41dfff?, 0x394e22915d48?})
2026-04-13T10:36:35.8085410Z 	/usr/local/go/src/net/fd_posix.go:68 +0x25
2026-04-13T10:36:35.8085746Z net.(*conn).Read(0x394e22980020, {0x394e22bfe001?, 0x41ddb4?, 0x39522dd56b68?})
2026-04-13T10:36:35.8086092Z 	/usr/local/go/src/net/net.go:196 +0x45
2026-04-13T10:36:35.8086351Z encoding/json.(*Decoder).refill(0x394e227a2000)
2026-04-13T10:36:35.8086650Z 	/usr/local/go/src/encoding/json/stream.go:167 +0x188
2026-04-13T10:36:35.8086953Z encoding/json.(*Decoder).readValue(0x394e227a2000)
2026-04-13T10:36:35.8087256Z 	/usr/local/go/src/encoding/json/stream.go:142 +0x85
2026-04-13T10:36:35.8087624Z encoding/json.(*Decoder).Decode(0x394e227a2000, {0x1308de0, 0x394e22dc8148})
2026-04-13T10:36:35.8088008Z 	/usr/local/go/src/encoding/json/stream.go:65 +0x75
2026-04-13T10:36:35.8088399Z net/rpc/jsonrpc.(*clientCodec).ReadResponseHeader(0x394e22dc8100, 0x394e22b5d560)
2026-04-13T10:36:35.8088818Z 	/usr/local/go/src/net/rpc/jsonrpc/client.go:76 +0x65
2026-04-13T10:36:35.8089102Z net/rpc.(*Client).input(0x394e22dd4060)
2026-04-13T10:36:35.8089359Z 	/usr/local/go/src/net/rpc/client.go:109 +0x9f
2026-04-13T10:36:35.8089777Z created by net/rpc.NewClientWithCodec in goroutine 1251
2026-04-13T10:36:35.8090088Z 	/usr/local/go/src/net/rpc/client.go:206 +0xb6
2026-04-13T10:36:35.8090265Z 
2026-04-13T10:36:35.8090468Z goroutine 1152 [select]:
2026-04-13T10:36:35.8090787Z github.com/google/syzkaller/vm/proxyapp.(*ProxyApp).doLogPooling.func1()
2026-04-13T10:36:35.8091362Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:324 +0x13b
2026-04-13T10:36:35.8091992Z created by github.com/google/syzkaller/vm/proxyapp.(*ProxyApp).doLogPooling in goroutine 108
2026-04-13T10:36:35.8092757Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient.go:314 +0x110
2026-04-13T10:36:35.8093143Z 
2026-04-13T10:36:35.8093221Z goroutine 729 [runnable]:
2026-04-13T10:36:35.8093460Z internal/poll.runtime_pollWait(0x7f4040991a00, 0x72)
2026-04-13T10:36:35.8093767Z 	/usr/local/go/src/runtime/netpoll.go:351 +0x85
2026-04-13T10:36:35.8094106Z internal/poll.(*pollDesc).wait(0x394e2291c700?, 0x394e22cf0000?, 0x0)
2026-04-13T10:36:35.8094486Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
2026-04-13T10:36:35.8094795Z internal/poll.(*pollDesc).waitRead(...)
2026-04-13T10:36:35.8095080Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
2026-04-13T10:36:35.8095444Z internal/poll.(*FD).Read(0x394e2291c700, {0x394e22cf0000, 0x800, 0x800})
2026-04-13T10:36:35.8095813Z 	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
2026-04-13T10:36:35.8096173Z net.(*netFD).Read(0x394e2291c700, {0x394e22cf0000?, 0x394e22cf0000?, 0x5?})
2026-04-13T10:36:35.8096532Z 	/usr/local/go/src/net/fd_posix.go:68 +0x25
2026-04-13T10:36:35.8096891Z net.(*conn).Read(0x394e2289a000, {0x394e22cf0000?, 0x7f3fc01c9f00?, 0x7f40888855d0?})
2026-04-13T10:36:35.8097258Z 	/usr/local/go/src/net/net.go:196 +0x45
2026-04-13T10:36:35.8097673Z crypto/tls.(*atLeastReader).Read(0x39523f0e1440, {0x394e22cf0000?, 0x394e228c6d20?, 0x394e6871f9d8?})
2026-04-13T10:36:35.8098123Z 	/usr/local/go/src/crypto/tls/conn.go:815 +0x3b
2026-04-13T10:36:35.8098458Z bytes.(*Buffer).ReadFrom(0x394e222042a8, {0x20313c0, 0x39523f0e1440})
2026-04-13T10:36:35.8098797Z 	/usr/local/go/src/bytes/buffer.go:229 +0x98
2026-04-13T10:36:35.8099196Z crypto/tls.(*Conn).readFromUntil(0x394e22204008, {0x2031680, 0x394e2289a000}, 0x394e6871fb98?)
2026-04-13T10:36:35.8099877Z 	/usr/local/go/src/crypto/tls/conn.go:837 +0xde
2026-04-13T10:36:35.8100185Z crypto/tls.(*Conn).readRecordOrCCS(0x394e22204008, 0x0)
2026-04-13T10:36:35.8100484Z 	/usr/local/go/src/crypto/tls/conn.go:626 +0x3db
2026-04-13T10:36:35.8100757Z crypto/tls.(*Conn).readRecord(...)
2026-04-13T10:36:35.8100996Z 	/usr/local/go/src/crypto/tls/conn.go:588
2026-04-13T10:36:35.8101310Z crypto/tls.(*Conn).Read(0x394e22204008, {0x394e22dc6001, 0x5ff, 0x1?})
2026-04-13T10:36:35.8101657Z 	/usr/local/go/src/crypto/tls/conn.go:1393 +0x145
2026-04-13T10:36:35.8101937Z encoding/json.(*Decoder).refill(0x394e2294e140)
2026-04-13T10:36:35.8102236Z 	/usr/local/go/src/encoding/json/stream.go:167 +0x188
2026-04-13T10:36:35.8102548Z encoding/json.(*Decoder).readValue(0x394e2294e140)
2026-04-13T10:36:35.8102848Z 	/usr/local/go/src/encoding/json/stream.go:142 +0x85
2026-04-13T10:36:35.8103223Z encoding/json.(*Decoder).Decode(0x394e2294e140, {0x1308d80, 0x394e228b4080})
2026-04-13T10:36:35.8103601Z 	/usr/local/go/src/encoding/json/stream.go:65 +0x75
2026-04-13T10:36:35.8103988Z net/rpc/jsonrpc.(*serverCodec).ReadRequestHeader(0x394e228b4060, 0x394e2287fe80)
2026-04-13T10:36:35.8104394Z 	/usr/local/go/src/net/rpc/jsonrpc/server.go:66 +0x6d
2026-04-13T10:36:35.8104778Z net/rpc.(*Server).readRequestHeader(0x394e22d2c0a0, {0x2058cb8, 0x394e228b4060})
2026-04-13T10:36:35.8105156Z 	/usr/local/go/src/net/rpc/server.go:589 +0x58
2026-04-13T10:36:35.8105506Z net/rpc.(*Server).readRequest(0x394e22d2c0a0?, {0x2058cb8, 0x394e228b4060})
2026-04-13T10:36:35.8105861Z 	/usr/local/go/src/net/rpc/server.go:549 +0x2f
2026-04-13T10:36:35.8106197Z net/rpc.(*Server).ServeCodec(0x394e22d2c0a0, {0x2058cb8, 0x394e228b4060})
2026-04-13T10:36:35.8106545Z 	/usr/local/go/src/net/rpc/server.go:464 +0x71
2026-04-13T10:36:35.8107048Z created by github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener.func1 in goroutine 230
2026-04-13T10:36:35.8107934Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:214 +0x2b2
2026-04-13T10:36:35.8108365Z 
2026-04-13T10:36:35.8108451Z goroutine 1253 [IO wait, 9 minutes]:
2026-04-13T10:36:35.8108721Z internal/poll.runtime_pollWait(0x7f4040c23600, 0x72)
2026-04-13T10:36:35.8109131Z 	/usr/local/go/src/runtime/netpoll.go:351 +0x85
2026-04-13T10:36:35.8109568Z internal/poll.(*pollDesc).wait(0x394e22ea2a00?, 0x900000036?, 0x0)
2026-04-13T10:36:35.8109949Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
2026-04-13T10:36:35.8110256Z internal/poll.(*pollDesc).waitRead(...)
2026-04-13T10:36:35.8110533Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
2026-04-13T10:36:35.8110821Z internal/poll.(*FD).Accept(0x394e22ea2a00)
2026-04-13T10:36:35.8111104Z 	/usr/local/go/src/internal/poll/fd_unix.go:613 +0x28c
2026-04-13T10:36:35.8111381Z net.(*netFD).accept(0x394e22ea2a00)
2026-04-13T10:36:35.8111619Z 	/usr/local/go/src/net/fd_unix.go:150 +0x29
2026-04-13T10:36:35.8111877Z net.(*TCPListener).accept(0x394e22eb0440)
2026-04-13T10:36:35.8112148Z 	/usr/local/go/src/net/tcpsock_posix.go:159 +0x1b
2026-04-13T10:36:35.8112420Z net.(*TCPListener).Accept(0x394e22eb0440)
2026-04-13T10:36:35.8112675Z 	/usr/local/go/src/net/tcpsock.go:387 +0x30
2026-04-13T10:36:35.8113092Z github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener.func1()
2026-04-13T10:36:35.8113757Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:210 +0x5f
2026-04-13T10:36:35.8114478Z created by github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener in goroutine 1252
2026-04-13T10:36:35.8115203Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:208 +0x291
2026-04-13T10:36:35.8115626Z 
2026-04-13T10:36:35.8115706Z goroutine 1256 [IO wait]:
2026-04-13T10:36:35.8115946Z internal/poll.runtime_pollWait(0x7f4040991600, 0x72)
2026-04-13T10:36:35.8116252Z 	/usr/local/go/src/runtime/netpoll.go:351 +0x85
2026-04-13T10:36:35.8116592Z internal/poll.(*pollDesc).wait(0x394e22ea3180?, 0x394e22f1a601?, 0x0)
2026-04-13T10:36:35.8116981Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:84 +0x27
2026-04-13T10:36:35.8117292Z internal/poll.(*pollDesc).waitRead(...)
2026-04-13T10:36:36.3082577Z make[1]: *** [Makefile:397: test] Error 1
2026-04-13T10:36:36.3085411Z make: *** [Makefile:325: presubmit_build] Error 2
2026-04-13T10:36:36.3086142Z 	/usr/local/go/src/internal/poll/fd_poll_runtime.go:89
2026-04-13T10:36:36.3086641Z internal/poll.(*FD).Read(0x394e22ea3180, {0x394e22f1a601, 0x5ff, 0x5ff})
2026-04-13T10:36:36.3087109Z 	/usr/local/go/src/internal/poll/fd_unix.go:165 +0x2ae
2026-04-13T10:36:36.3087561Z net.(*netFD).Read(0x394e22ea3180, {0x394e22f1a601?, 0x550c6e?, 0x12bbb20?})
2026-04-13T10:36:36.3087998Z 	/usr/local/go/src/net/fd_posix.go:68 +0x25
2026-04-13T10:36:36.3088364Z net.(*conn).Read(0x394e229000b8, {0x394e22f1a601?, 0x1?, 0x1?})
2026-04-13T10:36:36.3088755Z 	/usr/local/go/src/net/net.go:196 +0x45
2026-04-13T10:36:36.3089071Z encoding/json.(*Decoder).refill(0x394e22680640)
2026-04-13T10:36:36.3089720Z 	/usr/local/go/src/encoding/json/stream.go:167 +0x188
2026-04-13T10:36:36.3090142Z encoding/json.(*Decoder).readValue(0x394e22680640)
2026-04-13T10:36:36.3090539Z 	/usr/local/go/src/encoding/json/stream.go:142 +0x85
2026-04-13T10:36:36.3090995Z encoding/json.(*Decoder).Decode(0x394e22680640, {0x1308d80, 0x394e22eb4560})
2026-04-13T10:36:36.3091459Z 	/usr/local/go/src/encoding/json/stream.go:65 +0x75
2026-04-13T10:36:36.3091948Z net/rpc/jsonrpc.(*serverCodec).ReadRequestHeader(0x394e22eb4540, 0x394e22f060c0)
2026-04-13T10:36:36.3092443Z 	/usr/local/go/src/net/rpc/jsonrpc/server.go:66 +0x6d
2026-04-13T10:36:36.3092879Z net/rpc.(*Server).readRequestHeader(0x394e22e9c5f0, {0x2058cb8, 0x394e22eb4540})
2026-04-13T10:36:36.3093316Z 	/usr/local/go/src/net/rpc/server.go:589 +0x58
2026-04-13T10:36:36.3094018Z net/rpc.(*Server).readRequest(0x394e22e9c5f0?, {0x2058cb8, 0x394e22eb4540})
2026-04-13T10:36:36.3094452Z 	/usr/local/go/src/net/rpc/server.go:549 +0x2f
2026-04-13T10:36:36.3094852Z net/rpc.(*Server).ServeCodec(0x394e22e9c5f0, {0x2058cb8, 0x394e22eb4540})
2026-04-13T10:36:36.3095254Z 	/usr/local/go/src/net/rpc/server.go:464 +0x71
2026-04-13T10:36:36.3095850Z created by github.com/google/syzkaller/vm/proxyapp.makeMockProxyAppServerWithListener.func1 in goroutine 1253
2026-04-13T10:36:36.3096944Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/proxyappclient_tcp_test.go:214 +0x2b2
2026-04-13T10:36:36.3097449Z 
2026-04-13T10:36:36.3097540Z goroutine 23749075 [runnable]:
2026-04-13T10:36:36.3097782Z runtime.(*Func).Name(0x2bc72f0?)
2026-04-13T10:36:36.3098056Z 	/usr/local/go/src/runtime/symtab.go:810 +0xce
2026-04-13T10:36:36.3098382Z github.com/stretchr/testify/assert.CallerInfo()
2026-04-13T10:36:36.3098894Z 	/syzkaller/.cache/gomod/github.com/stretchr/testify@v1.11.1/assert/assertions.go:248 +0x2ae
2026-04-13T10:36:36.3099930Z github.com/stretchr/testify/mock.(*Mock).MethodCalled(0x394e22c33e00, {0x2433955, 0x8}, {0x39522e92f580, 0x2, 0x2})
2026-04-13T10:36:36.3100645Z 	/syzkaller/.cache/gomod/github.com/stretchr/testify@v1.11.1/mock/mock.go:558 +0xa0b
2026-04-13T10:36:36.3101425Z github.com/stretchr/testify/mock.(*Mock).Called(0x394e22c33e00, {0x39522e92f580, 0x2, 0x2})
2026-04-13T10:36:36.3102182Z 	/syzkaller/.cache/gomod/github.com/stretchr/testify@v1.11.1/mock/mock.go:491 +0x125
2026-04-13T10:36:36.3103304Z github.com/google/syzkaller/vm/proxyapp/mocks.(*ProxyAppInterface).PoolLogs(0x394e22c33e00, {}, 0x39523f0e14b8)
2026-04-13T10:36:36.3104091Z 	/__w/syzkaller/syzkaller/gopath/src/github.com/google/syzkaller/vm/proxyapp/mocks/ProxyAppInterface.go:383 +0xab
2026-04-13T10:36:36.3104789Z reflect.Value.call({0x394e228b4720?, 0x394e226020d8?, 0x13?}, {0x15f8fcf, 0x4}, {0x394e22137ef8, 0x3, 0x3?})
2026-04-13T10:36:36.3105253Z 	/usr/local/go/src/reflect/value.go:586 +0xf06
2026-04-13T10:36:36.3105658Z reflect.Value.Call({0x394e228b4720?, 0x394e226020d8?, 0x0?}, {0x394e22d4bef8?, 0x0?, 0x0?})
2026-04-13T10:36:36.3106066Z 	/usr/local/go/src/reflect/value.go:369 +0xb9
2026-04-13T10:36:36.3107068Z net/rpc.(*service).call(0x394e22c84b80, 0x394e22c33e50, 0x394e22f08000, 0x394e22f08010, 0x394e22e0c600, 0x394e22e801e0, {0x132f220?, 0x2fa7b40?, 0x0?}, {0x1245d80, ...}, ...)
2026-04-13T10:36:36.3108158Z 	/usr/local/go/src/net/rpc/server.go:383 +0x1f4
2026-04-13T10:36:36.3108555Z created by net/rpc.(*Server).ServeCodec in goroutine 1282
2026-04-13T10:36:36.3108877Z 	/usr/local/go/src/net/rpc/server.go:480 +0x3c5
2026-04-13T10:36:36.3109194Z FAIL	github.com/google/syzkaller/vm/proxyapp	601.048s
```

--- a/vm/proxyapp/proxyappclient_tcp_test.go
+++ b/vm/proxyapp/proxyappclient_tcp_test.go
@@ -179,6 +179,9 @@ func TestCtor_TCP_Reconnect_PoolChanged(t *testing.T) {
 		}).
 		Return(nil).
 		On("PoolLogs", mock.Anything, mock.Anything).
+		Run(func(args mock.Arguments) {
+			<-t.Context().Done()
+		}).
 		Return(nil)
 
 	p, _ := ctor(makeTestParams(), testTCPEnv(port))
@@ -211,11 +214,11 @@ func makeMockProxyAppServerWithListener(t *testing.T, l net.Listener) (*mockProx
 			if err != nil {
 				panic("failed to accept connection")
 			}
-			go server.ServeCodec(jsonrpc.NewServerCodec(conn))
-
 			connsMu.Lock()
 			conns = append(conns, conn)
 			connsMu.Unlock()
+
+			go server.ServeCodec(jsonrpc.NewServerCodec(conn))
 		}
 	}()
 

